### PR TITLE
Rewrite of GetTxBlock and GetTxBlockVerbose APIs to match spec

### LINKED ISSUE #1595: Implement gettxblockverbose

### LINKED ISSUE #1676: Z2-Converter API Mainnet : GetTxBlock
### Issue Summary:
Calling `GetTxBlock` on the ZQ1 and ZQ2 has the following diff.
Here are the URLs for ZQ1 and ZQ2.
ZQ2 : http://34.124.218.159:4201/
ZQ1 : http://34.124.185.137:4201/
### Result:
**Missing keys:**
```
Keys present in ZQ1 but missing in ZQ2:
  root['result']['header']['DSBlockNum']

Keys present in ZQ2 but missing in ZQ1:
  root['result']['header']['DsBlockNum']

```
**Different Keys:**
```
Values that differ between ZQ2 and ZQ1:
  root['result']['body']['BlockHash']:
    ZQ2: 7c7c754408e16f11b93921ce9ee6aa704cd3189e06d9b2e141efbd3ae0f93b35
    ZQ1: 424da7e159b0a4504ce307952b285b7eb6ed2693df41cc81a8e96862f58c4b9b
  root['result']['body']['HeaderSign']:
    ZQ2: 00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
    ZQ1: 0F2902579BD02FACE1778FDAAC4872000DD3C463A82405354CC342DABC64CD001D8DA143F0688A520E21E7382CEC7C366FA1D9F98C0B6B0F9A946CC4F67AA0B4
  root['result']['header']['MbInfoHash']:
    ZQ2: 0000000000000000000000000000000000000000000000000000000000000000
    ZQ1: 2f9dd0a266f5ef9a4294ee1f961b096e4af7f9da3cb878d88ee1e89a54ce8759
  root['result']['header']['MinerPubKey']:
    ZQ2: 0x000000000000000000000000000000000000000000000000000000000000000000
    ZQ1: 0x020177703339FDD3C56E3A5E4BAEDB253299C53BFC7A213679F9AD81ABA5BAB024
  root['result']['header']['NumMicroBlocks']:
    ZQ2: 0
    ZQ1: 4
  root['result']['header']['PrevBlockHash']:
    ZQ2: e29d9c1aefde7728284a240584045b33e67f54cd061611401d0313f910b4f038
    ZQ1: 4ab87748ca5aaf0dd451c01f877ff92b4dd918a8c31fdf95872ff90368d76a62
  root['result']['header']['StateDeltaHash']:
    ZQ2: 0000000000000000000000000000000000000000000000000000000000000000
    ZQ1: 4675069ff186f06d1af45a42bf65c1fe132612af6d8070b5cda8d30b0fad5bea
  root['result']['header']['StateRootHash']:
    ZQ2: 0297829e719795a14b56f5dcb875b7546c6a4d9e0b10190aecef256319cd3335
    ZQ1: 3a600ca11a0afb3ec7cbd3467e4d7d65ab7e6ba9514bf53e9d549e207cf07e6d
  root['result']['header']['Version']:
    ZQ2: 0
    ZQ1: 1

```
### Observations:
1) Fix the typo in ZQ2. Change `DsBlockNum` to  `DSBlockNum`.
2) Significant differences exist between various data fields in ZQ1 and ZQ2. There is already a discussion happening in Slack regarding the response difference.

###  Conclusion:
If the above differences are expected and align with the migration to ZQ2, no further action is required. However, if these differences are unexpected, they may indicate issues that must be addressed.

### PR DESCRIPTION
**Pull request by maxconway**
- Split GetTxBlock and GetTxBlockVerbose APIs into separate functions rather than trying to share code (This creates some redundancy but leaves room for future changes without creating bugs)
- Rewrite of docs of GetTxBlockVerbose to match observed return values from ZQ1
- Various changes to return types and values in both to fit observed return values from ZQ1
- Fixes bug introduced in #1752
- Add test for GetTxBlockVerbose
- Much more extensive tests for GetTxBlock

Closes #1676
Closes #1595
References #79

### CODE REVIEW

File: zilliqa/src/api/types/zil.rs [hunk 1]
```diff
@@ -33,12 +33,87 @@ pub struct TxBlock {
...
+                rewards: 0,
+                txn_fees: 0,
+                prev_block_hash: block.parent_hash().into(),
+                block_num: block.number(),
+                timestamp: block
+                    .timestamp()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .unwrap_or_default()
+                    .as_micros(),
+                mb_info_hash: B256::ZERO, // Appears obsolete in ZQ2
```
Comment (JamesHinshelwood) [inline zilliqa/src/api/types/zil.rs:53]: A bit nitpicky, but I'd prefer if the language of these comments was more assertive :) The current wording leads to some doubt when reading the code as to whether there is more work to do here to set this values differently.

I agree with all your observations and would be fine with just removing "appears" from them all :)

File: zilliqa/src/api/types/zil.rs [hunk 2]
```diff
@@ -33,12 +33,87 @@ pub struct TxBlock {
...
+    pub fn new(block: &Block) -> Self {
         let mut scalar = [0; 32];
         scalar[31] = 1;
         TxBlock {
             header: TxBlockHeader {
+                version: 1,                                    // To match ZQ1
+                gas_limit: ScillaGas::from(block.gas_limit()), // In Scilla
+                gas_used: ScillaGas::from(block.gas_used()),   // In Scilla
+                rewards: 0,
+                txn_fees: 0,
```
Comment (JamesHinshelwood) [inline zilliqa/src/api/types/zil.rs:45]: Is there an issue for these?

Comment (maxconway) [reply zilliqa/src/api/types/zil.rs]: There are now: #1823, #1824

File: zilliqa/src/api/types/zil.rs (hunk 1 above)
Comment (maxconway) [reply zilliqa/src/api/types/zil.rs]: Done

File: zilliqa/src/api/types/zil.rs [hunk 3]
```diff
@@ -79,11 +154,15 @@ impl TxBlock {
 
 #[derive(Clone, Serialize)]
 #[serde(rename_all = "PascalCase")]
-pub struct TxBlockHeader {
+pub struct TxBlockVerboseHeader {
     pub version: u8,
+    #[serde(with = "num_as_str")]
```
[the remark this answers is not in the document]
Comment (maxconway) [reply zilliqa/src/api/types/zil.rs:159]: I thought about this but my thinking was that it's quite nice to have API return types in Rust mirror the JSON return types closely, and I don't think the duplication is costing anything. Obviously there's the potential for a future bug if we want to add a field to GetTxBlock and forget to add it to GetTxBlockVerbose, but it's just as possible we might want it to only be in the non-verbose version and create a bug that way.

**Review by JamesHinshelwood — APPROVED** [review]

**Pull request merged**